apis/pubsub/v1alpha1/subscription_types.go

```diff
@@ -143,6 +147,26 @@ type PushConfig struct {
 	PushEndpoint string `json:"pushEndpoint,omitempty"`
 }
 
+// BigqueryConfig contains configuration for a bigquery delivery endpoint.
+type BigqueryConfig struct {
+	// Bigquery table to deliver messages to.
+	Table string `json:"table,omitempty"`
+
+	// When enabled, the topic schema will be used when writing to BigQuery. Else,
+	// tes the message bytes to a column called data in BigQuery.
+	UseTopicSchema bool `json:"useTopicSchema,omitempty"`
+
+	// When enabled, the metadata of each message is written to additional columns in
+	// the BigQuery table. Else, the metadata is not written to the BigQuery table.
+	// https://cloud.google.com/pubsub/docs/bigquery?hl=ru#write-metadata
+	WriteMetadata bool `json:"writeMetadata,omitempty"`
+
+	// When enabled along with the "Use topic schema" option, any field that is present in
+	// the topic schema but not in the BigQuery schema will be dropped. Else, messages with extra fields are not written
+	// and remain in the subscription backlog.
+	DropUnknownFields bool `json:"dropUnknownFields,omitempty"`
+}
```

Comment:
All these fields are optional, right?

So I think we should add the `// +optional` marker and change all types to be pointers.

Reply:
Agree, I will check.

Reply:
It works but it changes existing values to "false". I mean I had this:
```
bigqueryConfig:
      table: "..."
      useTopicSchema: true
      writeMetadata: true
      dropUnknownFields: false
```
changed to
```
bigqueryConfig:
      table: "..."
      useTopicSchema: false
```
and all 3 flags become disabled
<img width="464" alt="image" src="https://user-images.githubusercontent.com/5798108/196996023-8c6b6d4d-ec5f-42ba-9c17-fcd99c31e054.png">

Is it expected? I expected writeMetadata to stay enabled but maybe I'm wrong.

Reply:
I didn't get it. Do you mean that you specified `true` in the managed resource manifest and in the interface it was changed to Disabled (false)?